lib/src/lifecycle_module.dart

```diff
@@ -334,8 +334,14 @@ abstract class LifecycleModule extends SimpleModule
         _didLoadChildModuleController.add(childModule);
         completer.complete();
       } catch (error, stackTrace) {
-        await _didUnloadChildModuleSubscriptions[childModule]?.cancel();
-        await _willUnloadChildModuleSubscriptions[childModule]?.cancel();
+        StreamSubscription<LifecycleModule> didUnloadSub =
+            _didUnloadChildModuleSubscriptions.remove(childModule);
+        await didUnloadSub?.cancel();
```

Comment:
I saw that pattern below, but I found it harder to read (personally). If the change is desired, I can make it.